core/shared/src/test/scala/fs2/StreamSpec.scala

```diff
@@ -914,6 +914,99 @@ class StreamSpec extends Fs2Spec {
       }
     }
 
+    "evalFilter" - {
+      "with effectful const(true)" in {
+        val s = pureStreamGenerator[Int].sample
```

Comment:
Not decided on the approach - taking a single sample is hardly a nice test, but I can't use `forAll` on a stream with effects because the JS build complains about `unsafeRunSync()` (even when `SyncIO` is used!). I guess using a concrete example (or a few) will be better...

Reply:
Switch to `IO` instead of `SyncIO` and the tests should pass on Scala.js. You should also be able to use generators, just don't call `unsafeRunSync` -- `forAll { /* return F[Assertion] */ }`

Reply:
I did that, and (locally) I'm still getting the same issue... `cannot synchronously await result on JavaScript; use runAsync or unsafeRunAsync` (even though I never used anything with `unsafe` in these tests)